Added support for Keebio BDN9 and Cannon Keys db60

ambustion opened the pull request:
<!---
    For keyboard addition or changes, Vial will accept keyboards that implement "default" and "via" keymaps.
    For Vial-enabled keymaps please ensure that VIAL_INSECURE is not set.

    User keymaps (i.e. https://github.com/vial-kb/vial-qmk/tree/vial/users) will not be accepted at the moment.

    For other core changes, please explain what you are changing and why.

    Before submitting a PR, delete the entirety of this comment and document your changes.
-->

ambustion committed:
Add support for CannonKeys db60 and Keebio BDN9

ambustion committed:
remove db60 config from bdn9

xyzz commented on keyboards/cannonkeys/db60/keymaps/vial/rules.mk:
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
+TAP_DANCE_ENABLE = yes
+VIAL_INSECURE = yes
```
should be removed

xyzz commented on keyboards/keebio/bdn9/keymaps/vial/rules.mk:
```diff
@@ -0,0 +1,9 @@
+VIAL_ENABLE = yes
+VIAL_ENCODERS_ENABLE = yes
+ENCODER_ENABLE = yes
+VIAL_INSECURE = yes
```
same as above, should be removed

ambustion replied:
Is it the insecure part or the lines for encoders?

ambustion committed:
Removed suggested changes

ambustion committed:
updated a few keys for db60

xyzz commented:
- db60/keymaps/vial/rules.mk  is empty, vial isn't actually getting enabled
- db60/keymaps/vial/db60.jsonZone.Identifier shouldn't be in the pr
- db60/keymaps/vial/db60.json also shouldn't be in the pr

ambustion committed:
Removed erroneus db60.json and fixed rule.mk

ambustion committed:
Set unlock combo and removed rgb features not on board.

ambustion committed:
Added config for Keebio KBO-5000

Not totally sure if size of firmware is too close to the edge or how that impacts vial. 588 bytes free when building firmware.

ambustion committed:
Added security keys to keebio bdn9

xyzz commented:
the build is still failing

ambustion committed:
added RGB_Matrix_Enable

ambustion committed:
Update rules.mk for rev1 fw size limitations

xyzz commented on keyboards/keebio/bdn9/keymaps/vial/keymap.c:
```diff
@@ -0,0 +1,67 @@
+
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
any reason to have this here when you're using vial encoders?

ambustion replied:
No other than it was in the default keymap I was using.

xyzz commented on keyboards/keebio/bdn9/keymaps/vial/rules.mk:
```diff
@@ -0,0 +1,11 @@
+LTO_ENABLE = yes
+COMMAND_ENABLE = no
+ifdef RGB_MATRIX_ENABLE
```
While this is a valid syntax i really don't understand what you're trying to do here; e.g. why are you only enabling Vial when RGB matrix is enabled?

ambustion replied:
The build fails on the BDN9 rev1 due to firmware size so I used that as a check to enable only for the rev2. If there's a way to check for which rev is being built that would be more correct but I'm not sure how to do that.

xyzz replied:
If that's the case, can you put the keymap under `keebio/bdn9/rev2/keymaps/vial`? Otherwise, the build bot isn't even currently trying to build rev2 so it's kinda pointless to have vial support for it in vial-qmk.

ambustion commented:
Ok alright, I didn't realize that path structure was ok. Thank you for your
guidance.

Brendon Rathbone
C: 403.305.4443


On Sun, Dec 12, 2021 at 1:40 PM xyzz ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In keyboards/keebio/bdn9/keymaps/vial/rules.mk
> <https://github.com/vial-kb/vial-qmk/pull/80#discussion_r767324982>:
>
> > @@ -0,0 +1,11 @@
> +LTO_ENABLE = yes
> +COMMAND_ENABLE = no
> +ifdef RGB_MATRIX_ENABLE
>
> If that's the case, can you put the keymap under
> keebio/bdn9/rev2/keymaps/vial? Otherwise, the build bot isn't even
> currently trying to build rev2 so it's kinda pointless to have vial support
> for it in vial-qmk.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vial-kb/vial-qmk/pull/80#discussion_r767324982>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACVX7GQIK5U5FX3WSI7CXG3UQUCFDANCNFSM5IAPGTKQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

ambustion committed:
Moved bdn9 keymap to rev2/keymaps/vial as rev1 build has firmware size limitation
